vizro-core/src/vizro/models/_components/graph.py

```diff
@@ -81,3 +93,10 @@ def build(self):
             color="grey",
             parent_className="loading-container",
         )
+
+    @staticmethod
+    def _update_theme(fig: go.Figure, theme_selector: bool):
+        # Basically the same as doing fig.update_layout(template="vizro_light/dark") but works for both the call in
+        # self.__call__ and in the update_graph_theme callback.
+        fig["layout"]["template"] = themes.dark if theme_selector else themes.light
+        return fig
```

Comment:
I hope you're right that this part of the code is the same as doing: `fig.update_layout(template="vizro_light/dark")` even from the performances point of view (at least almost the same).

It's true assuming that the `pio` object is the `server` global variable that probably just change the key such as `"vizro_light"` with the value `themes.light` in some point of moment (don't exactly know when).

If the `pio` object makes these changes somehow in the `frontend` side (by storing the `pio` mapper on the client side and making `"vizro_light"` -> `themes.light` change on the frontend), then I guess we should go with this approach (`fig.update_layout(template="vizro_light/dark")`) because of performances.

Reply:
Yeah, this is something I already checked but you are absolutely right to think of it!

These things are exactly the same:
```
px.scatter([], template=vizro._themes.dark)
px.scatter([], template="vizro_dark")
px.scatter([]).update_layout(template=vizro._themes.dark)
px.scatter([]).update_layout(template="vizro_dark")
```

If you supply `template` as a string `"vizro_dark"`, all plotly does it look for key `"vizro_dark"` in the `pio.defaults` dictionary and return the corresponding `vizro._themes.dark` object (type `Template`, basically a dict). This is all done serverside, no frontend magic involved. So actually the new solution is very slightly more performant because we don't need to do the dictionary lookup (if possible I would have done it using the strings though, since they are "more public" than the internal `Template` objects).

What it means is that if we did want to make this a clientside callback, we basically have to store the JSON content of `vizro_dark` and `vizro_light` themes clientside in `dcc.Store` to switch between them.

Reply:
Yeah, I see. I just tested it and everything is super clear to me now. Thanks for the explanation 😄.

> What it means is that if we did want to make this a clientside callback, we basically have to store the JSON content of vizro_dark and vizro_light themes clientside in dcc.Store to switch between them.

Yes, that's the idea I have. Do we want to load client machine memory for a little (~50KB) and to optimise time execution when the theme is changed is the question we need to consider.